front-end/src/components/App.js

```diff
@@ -0,0 +1,102 @@
+import { useSelector, useDispatch } from "react-redux";
+import { changePage, fetchProducts } from "../actions";
+import { useEffect, useState } from "react";
+
+
+function App() {
+  // Both the product matrix and the pagination are rendered in this component
+  const dispatch = useDispatch();
+  const productData = useSelector(state => state.products);
+  const [page, setPage] = useState(1);
+  
+  useEffect(() => {
+    if (page !== undefined) {
+      console.log('useEffect - Page is now: ', page)
+    }
+  }, [page, productData])
+
+  const handlePageClick = (e) => {
+    if (productData.productCount < 10) {
+      setPage(1);
+    }
+    if(e.target.innerText === 'Previous') {
+      const prevPage = page - 1;
+      if (prevPage < 1) {
+        console.log('already on first page');
+        return null;
+      }
+      dispatch(changePage(prevPage, productData.categoryParam, productData.sortParam, productData.queryParam))
+      setPage(prevPage);
+    } else if (e.target.innerText === 'Next') {
+      const nextPage = page + 1;
+      //checks to see if the next page is greateer
+      if (nextPage > Math.ceil(productData.productCount/9)) {
+        console.log('already on last page');
+        return null;
+      }
+      dispatch(changePage(nextPage, productData.categoryParam, productData.sortParam, productData.queryParam))
+      setPage(nextPage);
+
+    } else {
+      //case for clicking on a page number
+      dispatch(changePage(Number(e.target.innerText), productData.categoryParam, productData.sortParam, productData.queryParam))
+      setPage(Number(e.target.innerText));
+    }
+  }
+
+  let currentProducts = [];
+  if(productData) {
+    currentProducts = productData.products;
+  }
+
+  const pageNumbers = []
+
+  for (let i = 1; i <= Math.ceil(productData.productCount / 9); i++) {
+    pageNumbers.push(i);
+  }
+
+  return (
+    <div>
+      <div className="container" id="products-container">
+        <div className="row row-cols-1 row-cols-sm-3 g-3">
+          {currentProducts.map((product, i) => {
+            let name = product.name
+            let category = product.category
```

Comment:
these variables aren't needed, could just use them below ie product.name instead of the name variable